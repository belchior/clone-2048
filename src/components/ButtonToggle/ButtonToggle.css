@import '../../static/vars.css';

@media all {
  .btn-toggle-label {
    display: inline-flex;
    align-items: center;
  }
  .btn-toggle-label .btn-toggle {
    margin-right: 0.5rem;
  }
  .btn-toggle {
    border-radius: 1rem 1rem 1rem 1rem;
    border: 1px solid var(--blue2);
    background-color: var(--blue1);
    width: 4rem;
    height: 2rem;
    transition: all 0.3s;
  }
  .btn-toggle:focus {
    outline: none;
  }
  .btn-toggle::before {
    content: '';
    border-radius: 50%;
    display: inline-block;
    width: 2rem;
    height: 2rem;
    background-color: var(--blue2);
    transform: translate3d(1rem, -1px, 0);
    transition: all 0.3s;
  }
  .btn-toggle[data-active="false"] {
    border-color: var(--gray3);
    background-color: var(--gray2);
  }
  .btn-toggle[data-active="false"]::before {
    background-color: var(--gray3);
    transform: translate3d(-1rem, -1px, 0);
  }
}
