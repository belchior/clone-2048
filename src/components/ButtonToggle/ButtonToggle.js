import React from 'react';
import PropTypes from 'prop-types';

import './ButtonToggle.css';

export const ButtonToggle = ({onClick, active = false, label = '', title = label}) => {
  return (
    <label htmlFor="toggleButton" className="btn-toggle-label">
      <button id="toggleButton" className="btn-toggle"
        onClick={onClick} data-active={active} title={title}
      ></button>
      {label}
    </label>
  );
};

ButtonToggle.propTypes = {
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool,
  label: PropTypes.string,
  title: PropTypes.string,
};
