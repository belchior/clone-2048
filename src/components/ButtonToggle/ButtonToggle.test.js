import React from 'react';
import { ButtonToggle } from './ButtonToggle';
import renderer from 'react-test-renderer';

const onClick = jest.fn();

it('ButtonToggle should render without crashing', () => {
  let tree = renderer.create(
    <ButtonToggle onClick={onClick} active={true} label="toggle button" title="my toggle button" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ButtonToggle should call the callback when click', () => {
  let tree = renderer.create(<ButtonToggle onClick={onClick} />).toJSON();
  tree.children[0].props.onClick();
  expect(onClick.mock.calls).toHaveLength(1);
});

it('ButtonToggle should assume data-active as false if active prop not passed', () => {
  let tree = renderer.create(<ButtonToggle onClick={onClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ButtonToggle should assume data-active as false if active prop is false', () => {
  let tree = renderer.create(<ButtonToggle onClick={onClick} active={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ButtonToggle should assume data-active as true if active prop is true', () => {
  let tree = renderer.create(<ButtonToggle onClick={onClick} active={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ButtonToggle should assume title value as the same as label if title are not passed', () => {
  let tree = renderer.create(<ButtonToggle onClick={onClick} label="my label" />).toJSON();
  expect(tree).toMatchSnapshot();
});
