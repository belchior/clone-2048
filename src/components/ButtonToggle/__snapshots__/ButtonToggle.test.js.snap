// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonToggle should assume data-active as false if active prop is false 1`] = `
<label
  className="btn-toggle-label"
  htmlFor="toggleButton"
>
  <button
    className="btn-toggle"
    data-active={false}
    id="toggleButton"
    onClick={[Function]}
    title=""
  />
  
</label>
`;

exports[`ButtonToggle should assume data-active as false if active prop not passed 1`] = `
<label
  className="btn-toggle-label"
  htmlFor="toggleButton"
>
  <button
    className="btn-toggle"
    data-active={false}
    id="toggleButton"
    onClick={[Function]}
    title=""
  />
  
</label>
`;

exports[`ButtonToggle should assume data-active as true if active prop is true 1`] = `
<label
  className="btn-toggle-label"
  htmlFor="toggleButton"
>
  <button
    className="btn-toggle"
    data-active={true}
    id="toggleButton"
    onClick={[Function]}
    title=""
  />
  
</label>
`;

exports[`ButtonToggle should assume title value as the same as label if title are not passed 1`] = `
<label
  className="btn-toggle-label"
  htmlFor="toggleButton"
>
  <button
    className="btn-toggle"
    data-active={false}
    id="toggleButton"
    onClick={[Function]}
    title="my label"
  />
  my label
</label>
`;

exports[`ButtonToggle should render without crashing 1`] = `
<label
  className="btn-toggle-label"
  htmlFor="toggleButton"
>
  <button
    className="btn-toggle"
    data-active={true}
    id="toggleButton"
    onClick={[Function]}
    title="my toggle button"
  />
  toggle button
</label>
`;
